util/libcrypto.num

```diff
@@ -5438,3 +5438,5 @@ ASYNC_set_mem_functions                 ?	3_1_0	EXIST::FUNCTION:
 ASYNC_get_mem_functions                 ?	3_1_0	EXIST::FUNCTION:
 BIO_ADDR_dup                            ?	3_1_0	EXIST::FUNCTION:SOCK
 CMS_final_digest                        ?	3_1_0	EXIST::FUNCTION:CMS
+ossl_strcasecmp                         ?	3_0_3	EXIST::FUNCTION:
+ossl_strncasecmp                        ?	3_0_3	EXIST::FUNCTION:
```

Comment:
Do we really want to add private functions as exported ones? Would it make sense to make these functions public?

Reply:
I got linker errors without this. I would prefer not exporting these functions.

Reply:
You'd have to link the object file with these functions to all the shared modules we produce and also to the apps in case it is dynamically linked. I do not think that is a good option. Basically the problem you have with the FIPS provider (where it is unavoidable) would replicate for all the other produced binaries.

Reply:
... except that we did some hard work not to have internal symbols exported like this (we had a large handful of those during the development of 3.0)

Reply:
Sure, but I think we need them available in apps, engines, libssl and tests

Reply:
I was pretty sure we had run into this problem before, so I went looking.  I suggest a study of ossl_sleep()...

Reply:
IMO we should discuss this with OTC if we want this as an exception, or just publish it as a public functions OPENSSL_str[n]casecmp.

Reply:
ossl_sleep() is static inline function. Unusable example IMO given this needs a global storage.

Reply:
We are using the approach with public functions. Thanks for the discussion!